src/ln/msgs.rs

```diff
@@ -382,9 +384,93 @@ impl NetAddress {
 			&NetAddress::OnionV3 {..} => { 4 },
 		}
 	}
+
+	fn len(&self) -> u16 {
+		match self {
+			&NetAddress::IPv4 { .. } => { 6 },
+			&NetAddress::IPv6 { .. } => { 18 },
+			&NetAddress::OnionV2 { .. } => { 12 },
+			&NetAddress::OnionV3 { .. } => { 37 },
+		}
+	}
 }
 
-#[derive(Clone)]
+impl Writeable for NetAddress {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ::std::io::Error> {
+		match self {
+			&NetAddress::IPv4 { ref addr, ref port } => {
+				0u8.write(writer)?;
```

Comment:
The types are 1-indexed as the 0 value used to be padding but has since been removed.

Reply:
Ooops, fixed